secrets/azurekeyvault/akv_test.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2019 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azurekeyvault
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"gocloud.dev/internal/testing/setup"
+	"gocloud.dev/secrets"
+	"gocloud.dev/secrets/driver"
+	"gocloud.dev/secrets/drivertest"
+
+	"github.com/Azure/azure-sdk-for-go/services/keyvault/v7.0/keyvault"
+	"github.com/Azure/go-autorest/autorest"
+)
+
+// Prerequisites for --record mode
+//
+// 1. Sign-in to your Azure Subscription at http://portal.azure.com.
+//
+// 2. Create a KeyVault, see https://docs.microsoft.com/en-us/azure/key-vault/quick-create-portal.
+//
+// 3. Choose an authentication model. This test uses Service Principal, see https://docs.microsoft.com/en-us/rest/api/azure/index#register-your-client-application-with-azure-ad.
+// For documentation on acceptable auth models, see https://docs.microsoft.com/en-us/azure/key-vault/key-vault-whatis.
+//
+// 4. Set your environment variables depending on the auth model selection.
+// For Service Principal, please set the following, see https://docs.microsoft.com/en-us/go/azure/azure-sdk-go-authorization.
+// - AZURE_TENANT_ID
+// - AZURE_CLIENT_ID
+// - AZURE_CLIENT_SECRET
+// - AZURE_ENVIRONMENT
+// - AZURE_AD_RESOURCE to https://vault.azure.net
+//
+// 5. Create/Import a Key. This can be done in the Azure Portal or by code.
+
+// 6. Update constants below to match your Azure KeyVault settings.
+
+const (
+	defaultAzureEnvironment = "AZUREPUBLICCLOUD"
```

Comment:
This is not used anywhere.

Reply:
Its there to aide recording. Essentially, you need to configure the environment variables as follow:
`func initEnv() {
	env, err := azure.EnvironmentFromName("AZUREPUBLICCLOUD")
	if err != nil {
		log.Fatalln(err)
	}

	os.Setenv("AZURE_TENANT_ID", "YOUR AZURE AD TENANT ID")
	os.Setenv("AZURE_CLIENT_ID", "CLIENT ID OF THE SERVICE PRINCIPAL")
	os.Setenv("AZURE_CLIENT_SECRET", "CLIENT SECRET OF THE SERVICE PRINCIPAL")
	os.Setenv("AZURE_ENVIRONMENT", env.Name)

	vaultEndpoint := strings.TrimSuffix(env.KeyVaultEndpoint, "/")
	os.Setenv("AZURE_AD_RESOURCE", vaultEndpoint)
}`

Call initEnv() in the beginning of newHarness(). The last two lines in the above snippet will most likely resolve your error '
azure.BearerAuthorizer#WithAuthorization'

I am going to include this code in my next commit with some added instructions